src/components/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,343 @@
+  const {
+    label,
+    required,
+    useDatePicker,
+    dateValue,
+    helperText,
+    validationState,
+    validationText,
+    validationClassName,
+    validationStyle,
+    getDateValue,
+    closeClicked,
+    id,
+    isRange,
+    rangeConfig,
+    onConfirmRange,
+  } = props;
+  const [date, setDate] = useState<any>({ mm: "", dd: "", yyyy: "" });
+  const [dateError, setDateError] = useState("");
+  const [datePickerDate, setDatePickerDate] = useState("");
+  const [showPicker, setShowPicker] = useState(false);
+  const [range, setRange] = useState<RangeConfigProps>({
+    start: "",
+    end: "",
+    min: "",
+    max: "",
+  });
+
+  // if a date value is given as input then set that in the state
+  // * The format is important "mm/dd/yyyy"
+  useEffect(() => {
+    if (dateValue) {
+      let dateVal: any = new Date(dateValue);
+      let stateDateObj = { ...date };
+      stateDateObj.mm =
+        dateVal.getMonth() + 1 < 10
+          ? "0" + (dateVal.getMonth() + 1).toString()
+          : (dateVal.getMonth() + 1).toString();
+      stateDateObj.dd =
+        dateVal.getDate() < 10
+          ? "0" + dateVal.getDate().toString()
+          : dateVal.getDate().toString();
+      stateDateObj.yyyy = dateVal.getFullYear().toString();
+      setDate(stateDateObj);
+    }
+    if (isRange && rangeConfig) {
+      setRange(rangeConfig);
+    }
+  }, []);
+
+  const handleDateInput = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+    dateKey: string,
+    index: number
+  ) => {
+    let stateDateObj: any = { ...date };
+    event.target.value = event.target.value.replace(/[^0-9]+/g, "");
+    // if the given input has a length on 2 charcters and its not the last input field
+    if (event.target.value.length == 2 && index != 2) {
+      // get the next document and focus
+      let nextInput = document.getElementById(`date-input-${index + 1}-${id}`);
+      nextInput?.focus();
+    }
+
+    if (index == 2 && event.target.value.length > 4) {
+      return;
+    }
+    // set the data in the state
+    stateDateObj[dateKey] = event.target.value;
+    setDate(stateDateObj);
+    // if all inputs are filled then execute the local date validation function
+    if (stateDateObj.mm && stateDateObj.dd && stateDateObj.yyyy) {
+      defaultErrorValidation(stateDateObj);
+    }
+    getDateValue?.(
+      new Date(`${stateDateObj.mm}/${stateDateObj.dd}/${stateDateObj.yyyy}`)
+    );
+  };
+
+  // Function to handle cancel button click
+  const handleCancelClick = () => {
+    let stateDateObj = { ...date };
+    stateDateObj.mm = "";
+    stateDateObj.dd = "";
+    stateDateObj.yyyy = "";
+    setDate(stateDateObj);
+    // empty state value and error text
+    setDateError("");
+    getDateValue?.("");
+  };
+
+  // a default date validation that peforms basic checks
+  const defaultErrorValidation = (stateDateObj: any) => {
+    if (stateDateObj.mm && (stateDateObj.mm == 0 || stateDateObj.mm > 12)) {
```

Comment:
Consider using the strict equality operator (===) instead of the loose equality operator (==) in the conditional statement.

Reply:
Changes made 👍